frontend/front/src/pages/Admin/nav/Nav.js

```diff
@@ -8,12 +8,17 @@ import {
   useTheme,
 } from "@mui/material";
 
-import { Link } from "react-router-dom";
-import React from "react";
+import { Link, useNavigate } from "react-router-dom";
+import React, { useCallback } from "react";
 import { useLogoutUserMutation } from "../../../api/apiSlice";
 
 const Nav = () => {
+  const navigate = useNavigate();
   const [logout] = useLogoutUserMutation();
+  const handleLogout = useCallback(async () => {
+    await logout();
```

Comment:
Are the mutation functions that rtk query returns actual Promises? I had to do `mutation().unwrap()` elsewhere to get it to actually await.

Reply:
The mutation trigger (in this case, `logout`) returns a promise.
https://redux-toolkit.js.org/rtk-query/usage/mutations#frequently-used-mutation-hook-return-values